EIPS/eip-7928.md

```diff
@@ -0,0 +1,199 @@
+```
+
+The `BlockAccessList` is a deduplicated list of accessed addresses. For each address, it MUST contain a list of accessed storage keys. 
+For writes, each `SlotAccess` MUST contain an ordered list of transaction indices that accessed this key, and the final storage value after the last access. Transactions with writes that do not change the storage value MUST NOT contain a `value_after`.
+For contract deployments, the `value_after` MUST contain the runtime bytecode of the newly deployed contract.
+
+For reads, each `SlotAccess` MUST contain an empty `accesses` list.
+
+The `BalanceDiffs` structure tracks every address with a balance change, including transaction senders, recipients, and the block's coinbase address. 
+Each entry MUST include the transaction index and the signed balance delta per address for each transaction.
+11 bytes are sufficient to represent the total ETH supply.
+
+The `NonceDiffs` structure MUST record the post-block nonce values for all `CREATE2` deployer accounts and the deploayed contracts in the block.
+
+### State Transition Function
+
+Modify the state transition function to validate the block-level access lists:
+
+```python 
+def state_transition(block: Block) -> None:
+    computed_access_list = {}
+    computed_balance_diffs = {}
+    computed_nonce_diffs = {}
+
+    for idx, tx in enumerate(block.transactions):
+        # Execute transaction and track state accesses
+        accessed_items, balance_changes = execute_transaction(tx)
+        
+        # Process storage accesses
+        for (address, slot), is_write, value in accessed_items:
+            if (address, slot) not in computed_access_list:
+                computed_access_list[(address, slot)] = []
+            
+            access_entry = {"tx_index": idx}
+            if is_write:
+                access_entry["value_after"] = value
+            
+            computed_access_list[(address, slot)].append(access_entry)
+        
+        # Process balance changes
+        for address, delta in balance_changes:
+            if address not in computed_balance_diffs:
+                computed_balance_diffs[address] = []
+            computed_balance_diffs[address].append({"tx_index": idx, "delta": delta})
+
+    # Validate access list and balance diffs
+    if not validate_access_list(block.block_access_list, computed_access_list):
+        raise InvalidBlock("Mismatch in block-level access list.")
+        
+    if not validate_balance_diffs(block.balance_diffs, computed_balance_diffs):
+        raise InvalidBlock("Mismatch in balance diffs.")
+        
+    if not validate_nonce_diffs(block.nonce_diffs, computed_nonce_diffs):
+        raise InvalidBlock("Mismatch in nonce diffs.")
+```
+
+The BAL MUST be complete and accurate. It MUST NOT contain too few entries (missing accesses) or too many entries (spurious accesses). Any missing or extra entries in the access list, balance diffs, or nonce diffs SHALL result in block invalidation.
```

Comment:
I'm not sure what the correct term is but I think "complete" covers it: all items in the list are touched within the block, and items in the block which are touched are also in the list.